### ISSUE TITLE
Sandboxing eval

### ISSUE THREAD

**Issue by jtrakk** [cmcaine/Sandboxes.jl#3]
I see there are some caveats here:

"if the user imports any code with anything other than using X: something_safe. Users will need to be cautious of eval available on most module objects, for example."

Is it possible to prevent these threats too?

**Comment by cmcaine** [comment]
I'm going to open a PR on Julia for extra-bare baremodules that won't
import even Core, so that will deal with the first part.

For the second, I think documentation on how to use the sandbox will be
sufficient.

Do you have an interest in using Sandboxes? Knowing your use case could
help inform the design.

**Comment by cmcaine** [comment]
And I have a plan for making it easier to import the safe parts of modules
into the sandbox, too, but it's a bit complicated.

I might be applying for funding soon for expanding it.

**Comment by jtrakk** [comment]
This is entirely speculative, I don't have any investment in this kind of project, but: I think it would be very cool to have an [object-capability](https://github.com/dckc/awesome-ocap) system in julia. So I can restrict the capabilities of functions I call, and they can restrict the functions they call, etc.

**Comment by cmcaine** [comment]
Yes, capabilities are precisely a motivation for this :)

**Comment by jtrakk** [comment]
I wonder what other restrictions can be applied from within julia. Like all the cgroupsv2 and nsjail things: network access, filesystem access, memory limits, etc.

**Comment by cmcaine** [comment]
Yep, that's on the cards. Tho, as far as I know, all of those restrictions apply at the process level, but Julia can start new processes with OS-level restrictions like those, so long as it has the permission to do so.

**Comment by jtrakk** [comment]
I don't exactly understand this but maybe it's possible to have different permissions for different threads?
 https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v2.html#threads
